src/parse_nlp.jl

```diff
@@ -167,6 +167,10 @@ function _parse_NL_expr(m, x, tapevar, parent, values)
 
 end
 
+function _parse_NL_expr_runtime(m::Model, x::Complex, tape, parent, values)
+    error("Complex numbers are not supported in nonlinear expressions")
+end
+
 function _parse_NL_expr_runtime(m::Model, x::Number, tape, parent, values)
```

Comment:
Can we achieve the same result by changing `Number` to `Real` in this signature? There's a fallback below (`_parse_NL_expr_runtime(m::Model, x, tape, parent, values)`) that would catch complex numbers then.